pybrake/routes.py

```diff
@@ -43,28 +44,33 @@ def __init__(self, project_id, project_key, host):
   def _init(self):
     if self._stats is None:
       self._stats = {}
-      self._thread = Timer(self._flush_period, self._flush)
+      self._thread = Timer(self._flush_period, self.flush)
       self._thread.start()
 
-  def inc_request(self, method='', route='', status_code=0, time=None, ms=0):
-    self._init()
-
-    statKey = route_stat_key(method, route, status_code, time)
-    with self._lock:
-      if statKey in self._stats:
-        stat = self._stats.get(statKey)
-        stat.add(ms)
-      else:
-        self._stats[statKey] = RouteStat(method, route, status_code, time, ms)
-
-  def _flush(self):
+  def flush(self):
     stats = dict()
     with self._lock:
       stats = self._stats
       self._stats = None
+      self._thread.cancel()
+      self._thread = None
 
     if not stats:
       raise ValueError("Stats is empty")
 
     stats_json = json.dumps({"routes": [stat.__dict__ for _, stat in stats.items()]})
     requests.post(self._api_url, data=stats_json, headers=self._airbrake_headers)
+
+  def inc_request(self, method='', route='', status_code=0, time=None, ms=0):
+    self._init()
+
+    tm = datetime.utcnow().replace(second=0, microsecond=0).strftime('%Y-%m-%dT%H:%M:%SZ')
+    statKey = route_stat_key(method, route, status_code, tm)
+    with self._lock:
+      if statKey in self._stats:
+        stat = self._stats.get(statKey)
+        stat.add(ms)
+      else:
+        routeStat = RouteStat(method, route, status_code, time)
```

Comment:
3 lines above this is called `stat`, not `routeStat`. Unify names and move `Stat.add(ms)` outside if/else